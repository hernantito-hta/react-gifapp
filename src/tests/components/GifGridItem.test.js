import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from '../../components/GifGridItem';

describe('Testing components GifGridItem', () => {

    const title = 'Title description';
    const url = 'https://www.apitest.com/test';
    const wrapper = shallow(<GifGridItem url={url} title={title} />);

    test('should it validate the component', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should validate the title', () => {
        const parraf = wrapper.find('p');
        expect(parraf.text().trim()).toBe(title);
    })

    test('should validate the url', () => {
        const img = wrapper.find('img');
        expect(img.props().src).toBe(url);
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })
    
    test('should have animate__fadeIn', () => {
        const div = wrapper.find('div');
        expect(div.props().className).toContain('animate__fadeIn');
        expect(div.props().className.includes('animate__fadeIn')).toBe(true);
    })
    
})
