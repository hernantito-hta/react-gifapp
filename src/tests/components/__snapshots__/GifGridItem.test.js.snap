// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing components GifGridItem should it validate the component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Title description"
    src="https://www.apitest.com/test"
  />
  <p>
    Title description
  </p>
</div>
`;
